models/webhook_telegram.go

```diff
@@ -0,0 +1,321 @@
+// Copyright 2018 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"code.gitea.io/git"
+	api "code.gitea.io/sdk/gitea"
+)
+
+type (
+	// TelegramPayload represents
+	TelegramPayload struct {
+		Message   string `json:"text"`
+		ParseMode string `json:"parse_mode"`
+	}
+
+	// TelegramMeta contains the telegram metadata
+	TelegramMeta struct {
+		BotToken string `json:"bot_token"`
+		ChatID   string `json:"chat_id"`
+	}
+)
+
+// SetSecret sets the telegram secret
+func (p *TelegramPayload) SetSecret(_ string) {}
+
+// JSONPayload Marshals the TelegramPayload to json
+func (p *TelegramPayload) JSONPayload() ([]byte, error) {
+	p.ParseMode = "HTML"
+	p.Message = strings.Replace(p.Message, "<!--", "\\<!--", -1)
+	data, err := json.MarshalIndent(p, "", "  ")
+	if err != nil {
+		return []byte{}, err
+	}
+	return data, nil
+}
+
+func getTelegramCreatePayload(p *api.CreatePayload) (*TelegramPayload, error) {
+	// created tag/branch
+	refName := git.RefEndName(p.Ref)
+	title := fmt.Sprintf(`[<a href="%s">%s</a>] %s <a href="%s">%s</a> created`, p.Repo.HTMLURL, p.Repo.FullName, p.RefType,
+		p.Repo.HTMLURL+"/src/"+refName, refName)
+
+	return &TelegramPayload{
+		Message: title,
+	}, nil
+}
+
+func getTelegramDeletePayload(p *api.DeletePayload) (*TelegramPayload, error) {
+	// created tag/branch
+	refName := git.RefEndName(p.Ref)
+	title := fmt.Sprintf(`[<a href="%s">%s</a>] %s <a href="%s">%s</a> created`, p.Repo.HTMLURL, p.Repo.FullName, p.RefType,
```

Comment:
Replace `created` with `deleted` in code and comment.

Reply:
done